server/src/frameworks/Hardhat/HardhatProject.ts

```diff
@@ -253,6 +247,37 @@ export class HardhatProject extends Project {
     return resolveActionsFor(this.serverState, diagnostic, document, uri);
   }
 
+  private _sendToChild(message: Message) {
+    this._assertWorkerExists();
+    this._assertWorkerConnected();
+
+    this.workerProcess.send(message, (error) => {
+      if (error) {
+        this.workerStatus = WorkerStatus.ERRORED;
+        this.logger.error(`Error sending message to hardhat worker: ${error}`);
+
+        if ("requestId" in message) {
+          const messageWithId: { requestId: number } = message as any;
+          this._callErrorForRequest(messageWithId.requestId, error);
```

Comment:
I enhanced this to replicate the cleanup logic we use for an error message against a specific request id.
If there is no `requestId` we fall back as before.